### PR TITLE
update: updated Kotlin Roadmap

### PR DESCRIPTION
**Pull request by katia-energizer**
* Updated the roadmap as of January 2021.
* Added a changelog
* Moved roadmap notes to the bottom, after the roadmap table.
* Added a link to the YT board.
* Added a TOC.

### CODE REVIEW

File: pages/roadmap.md [hunk 1]
```diff
@@ -286,12 +220,82 @@ This table describes the biggest projects that we’re working on.
...
     </tr>
     <tr>
-        <td><strong>Ktor</strong>
+        <td><strong>Website</strong></td>
+        <td>
+            <ul>
+                <li><a href="https://youtrack.jetbrains.com/issue/KT-44338" target="_blank">Revamp Kotlin documentation</a></li>
+                <li><a href="https://youtrack.jetbrains.com/issue/KT-44339" target="_blank">Make the Kotlin website mobile friendly</a></li>
+                <li><a href="https://youtrack.jetbrains.com/issue/KT-44340" target="_blank">Design a new Kotlin visual style</a></li>
+                <li><a href="https://youtrack.jetbrains.com/issue/KT-44342" target="_blank">Close <em>try.kotlinlang.org</em></a></li>
```
Comment (koshachy) [inline pages/roadmap.md:229]: Nit-picking: maybe another verb with more straightforward meaning: disable or shut down?

File: pages/roadmap.md [hunk 2]
```diff
@@ -286,12 +220,82 @@ This table describes the biggest projects that we’re working on.
...
+
+* ✅ Compiler core: [Bootstrap the new compiler (make the new Kotlin compiler compile itself)](https://youtrack.jetbrains.com/issue/KT-42285)
+* ✅ Kotlin/Native: [Runtime performance: improve object allocation times](https://youtrack.jetbrains.com/issue/KT-42295)
+* ✅ Multiplatform: [KMM plugin: Fix major bugs](https://youtrack.jetbrains.com/issue/KT-42299)
+* ✅ Multiplatform: [KMM plugin: Run common tests on Android devices](https://youtrack.jetbrains.com/issue/KT-42298) (partially)
+* ✅ Libraries: [Improve `kotlinx-coroutines`(release v1.4)](https://youtrack.jetbrains.com/issue/KT-42317)
+
+### Postponed items
+
+We've decided to **postpone** the following items from the previous roadmap:
```
Comment (koshachy) [inline pages/roadmap.md:264]: Minor: previous version of the roadmap ?

File: pages/roadmap.md [hunk 3]
```diff
@@ -45,7 +39,15 @@ The goal of this roadmap is to give you a big picture. Here’s a list of our ke
 
 ## Kotlin roadmap by subsystem
 
-This table describes the biggest projects that we’re working on.
+You can view the biggest projects that we’re working on the [YouTrack board](https://youtrack.jetbrains.com/agiles/153-1251/current) and in the [Roadmap details](#roadmap-details) table.
```
Comment (koshachy) [inline pages/roadmap.md:42]: It seems like the other one "on" preposition is omitted ("working on on"). 
But it's better to rewrite the sentence to escape preposition doubling, provide a better reading, and relate it to the heading.

>The significant projects that we're working on are sorted into several subsystems. You can view them on the YouTrack board and in the Roadmap details table.

>We have sorted the significant projects that we're working on into several subsystems. You can view them on the YouTrack board and in the Roadmap details table.

File: pages/roadmap.md [hunk 4]
```diff
@@ -45,7 +39,15 @@ The goal of this roadmap is to give you a big picture. Here’s a list of our ke
...
+
+If you have any questions or feedback about the roadmap or the items on it, feel free to post them to YouTrack tickets or in the [#kotlin-roadmap](https://kotlinlang.slack.com/archives/C01AAJSG3V4) channel of Kotlin Slack ([request an invite](https://surveys.jetbrains.com/s3/kotlin-slack-sign-up)).
```
Comment (koshachy) [inline pages/roadmap.md:44]: _corresponding YouTrack tickets_ or a link to create the new one?

**Review by koshachy — APPROVED** [review]
Overall is fine 👍  

There are some minor issues. It's up to you to fix them or not.

File: pages/roadmap.md [hunk 5]
```diff
@@ -286,12 +220,82 @@ This table describes the biggest projects that we’re working on.
         </td>
     </tr>
     <tr>
-        <td><strong>Ktor</strong>
+        <td><strong>Website</strong></td>
+        <td>
+            <ul>
+                <li><a href="https://youtrack.jetbrains.com/issue/KT-44338" target="_blank">Revamp Kotlin documentation</a></li>
+                <li><a href="https://youtrack.jetbrains.com/issue/KT-44339" target="_blank">Make the Kotlin website mobile friendly</a></li>
```
[the remark this answers is not in the document]
Comment (katia-energizer) [reply pages/roadmap.md:227]: No, not in this case.

File: pages/roadmap.md (hunk 2 above)
Comment (katia-energizer) [reply pages/roadmap.md:264]: I don't want to use `version` for the roadmap. If it's unclear for users, I'll update phrasing next time. Copyeditiors are OK with the current phrasing.

File: pages/roadmap.md (hunk 3 above)
Comment (katia-energizer) [reply pages/roadmap.md]: We've updated with copyeidtors: 
> To view the biggest projects we're working on, visit the [YouTrack board](https://youtrack.jetbrains.com/agiles/153-1251/current) or the [Roadmap details](#roadmap-details) table.

File: pages/roadmap.md (hunk 1 above)
Comment (katia-energizer) [reply pages/roadmap.md:229]: It requires updating in 4 places including YT. I would keep it `Close`. Copyeditors are OK.

File: pages/roadmap.md (hunk 4 above)
Comment (katia-energizer) [reply pages/roadmap.md:44]: Add a link to a list of YT tickets.

File: pages/roadmap.md [hunk 6]
```diff
@@ -54,125 +56,89 @@ This table describes the biggest projects that we’re working on.
...
+        <td><strong>Language</strong></td>
         <td>
             <ul>
-                <li><p>Support JVM records<br/>(<a href="https://youtrack.jetbrains.com/issue/KT-42430" target="_blank">KT-42430</a>)</p></li>
-                <li><p>Support JVM sealed classes<br/>(<a href="https://youtrack.jetbrains.com/issue/KT-42433" target="_blank">KT-42433</a>)</p></li>
-                <li><p>Release inline classes as stable, secure Valhalla compatibility<br/>(<a href="https://youtrack.jetbrains.com/issue/KT-42434" target="_blank">KT-42434</a>)</p></li>
-                <li><p>Prototype multiple receivers<br/>(<a href="https://youtrack.jetbrains.com/issue/KT-42435" target="_blank">KT-42435</a>)</p></li>
+                <li><a href="https://youtrack.jetbrains.com/issue/KT-42430" target="_blank">Support JVM records</a></li>
+                <li><a href="https://youtrack.jetbrains.com/issue/KT-42433" target="_blank">Support JVM sealed classes</a></li>
+                <li><a href="https://youtrack.jetbrains.com/issue/KT-42434" target="_blank">Release inline classes as stable, secure Valhalla compatibility</a></li>
```
[the remark this answers is not in the document]
Comment (katia-energizer) [reply pages/roadmap.md:64]: Thx. Updated

**Comment by etolstoy** [comment]
👍 from my side

**Review by etolstoy — APPROVED** [review]

**Pull request merged**